state/executor.go

```diff
@@ -137,6 +137,17 @@ func (e *Executor) GetDumpTree(dump *Dump, parentHash types.Hash,
 	return next, nil
 }
 
+// Verbosity sets the log verbosity ceiling.
+func (e *Executor) Verbosity(level int) (string, error) {
+	if level < int(hclog.NoLevel) || level > int(hclog.Off) {
+		return hclog.Level(level).String(), fmt.Errorf("invalid log level: %d", level)
+	}
+
+	e.logger.SetLevel(hclog.Level(level))
```

Comment:
You are probably right, all files use the "Named(name string) Logger" method to create the logger. I guess the method doesn't work by reference, but by value... I'll have to look into that.

Reply:
However, it looks like it will work, when the logger is instantiated in the options, the following attribute is not set, causing the logging level to go by address...:

 // IndependentLevels causes subbloggers to be created with independent
 // level copy of this logger. That means you use SetLevel for this
 // logger will not affect subbloggers and SetLevel will not affect subbloggers
 // will not affect parents or siblings.
 IndependentLevels bool

I will certainly protest further...